pennylane/interfaces/torch.py

```diff
@@ -16,13 +16,48 @@
 compatible quantum nodes.
 """
 # pylint: disable=redefined-outer-name,arguments-differ
+from collections import Iterable
 import inspect
 from functools import partial
+import numbers
 
 import numpy as np
 import torch
+from torch.autograd.function import once_differentiable
 
-from pennylane.utils import unflatten
+from pennylane.utils import _flatten
+
+
+def unflatten_torch(flat, model):
```

Comment:
I wonder if one day we could unite all the `unflatten`s, since most of the functionality is quite similar except for checking for instances of `torch.Tensor` etc.

Reply:
Even though there is a bit of code duplication, I tend to find it makes more sense to have them separate, for a couple of reasons:

* For the 'core' parts of PennyLane, all optional ML dependencies should be relegated to the `interfaces` module, and sandboxed from each other; i.e., we shouldn't have TF/Torch/Autograd code in `utils`, `qnodes`, `operations` etc.

* There is very little use case for an unflatten function that can work with models containing PyTorch, TF, _and_ autograd instances. A user can only use one ML library per QNode.

* There is also differences in function and method names (`torch.Tensor.view()` vs `tf.reshape()`, `torch.Tensor.numel()` vs `tf.size()`), as well as differences in behaviour (`tf.size()` returns a `tf.Tensor[int]`, whereas `torch.numel()` returns a Python int).